nifi-api/src/main/java/org/apache/nifi/components/Allowable.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.components;
+
+public interface Allowable {
+
+    String getDisplayName();
+    String getDescription();
+}

```

Comment:
There are already some comments about the name of the interface but I would go a little bit further.
I would elevate it from being a purely technical tool to a domain entity, like this:

```suggestion
public interface PropertyValue {
    String getValue();
    String getDisplayName();
    String getDescription();
}
```
With this the interface represents a real-life concept instead of just being a convenient device.

(This would actually also allow us to use the new interface in the PropertyDescriptor but that would require a more careful analysis.)